dask/dataframe/io/parquet/arrow.py

```diff
@@ -108,523 +130,292 @@ def _index_in_schema(index, schema):
-                if column.statistics:
-                    cmin = column.statistics.min
-                    cmax = column.statistics.max
-                    cnull = column.statistics.null_count
-                    last = cmax_last.get(name, None)
-                    if no_filters:
-                        # Only think about bailing if we don't need
-                        # stats for filtering
-                        if cmin is None or (last and cmin < last):
-                            # We are collecting statistics for divisions
-                            # only (no filters) - Column isn't sorted, or
-                            # we have an all-null partition, so lets bail.
-                            #
-                            # Note: This assumes ascending order.
-                            #
-                            gather_statistics = False
-                            file_row_group_stats = {}
-                            file_row_group_column_stats = {}
-                            break
+            ]
 
-                    if single_rg_parts:
-                        to_ts = column.statistics.logical_type.type == "TIMESTAMP"
-                        s["columns"].append(
-                            {
-                                "name": name,
-                                "min": cmin if not to_ts else pd.Timestamp(cmin),
-                                "max": cmax if not to_ts else pd.Timestamp(cmax),
-                                "null_count": cnull,
-                            }
-                        )
+            # Row group count before (`num_row_groups_i`) and
+            # after (`num_row_groups`) filtering
+            num_row_groups_f = len(filtered_row_groups)
+            num_row_groups_i = len(file_frag.row_groups)
+
+            if split_row_groups:
+                # Splitting by row-group.
+                k = int(split_row_groups)
+                if k >= num_row_groups_f:
+                    # Split is larger than the number of row-groups in the file.
+                    if num_row_groups_f < num_row_groups_i:
+                        # 1+ row-groups are filtered - Need new fragment.
+                        metadata.append(_frag_subset(file_frag, filtered_row_groups))
                     else:
-                        cstats += [cmin, cmax, cnull]
-                    cmax_last[name] = cmax
+                        # Nothing was filtered - Use original fragment.
+                        metadata.append(file_frag)
                 else:
+                    # Splits are smaller than the number of row-groups in the file.
+                    # We will need to create multiple new fragments.
+                    for rg in range(0, num_row_groups_f, k):
+                        new_row_groups = [
+                            i for i in range(rg, rg + k) if i < num_row_groups_f
+                        ]
+                        metadata.append(_frag_subset(file_frag, new_row_groups))
```

Comment:
I would add the special case for `split_row_groups=True` (so `k` being 1), at least for now. Because then you can actually use the actual splittled row-group fragments (the ones in `filtered_row_groups`), and you don't need to subset the original `file_frag` (reason being: manually subsetting (for now, but have a PR to work on that) looses the statistics, while the ones splitted by row groups preserve the statistics)

Something like:

```
if k == 1:
    metadata.extend(filtered_row_groups)
```

Reply:
Very good point! (sorry I missed this)